### DISCUSSION TITLE
文字识别时EXE可以，但同样的代码放在DLL里出错

### DISCUSSION THREAD

**Discussion by jwj76** [PaddlePaddle/PaddleOCR#13823] [Q&A]
1、调用
std::vector<OCRPredictResult> ocr_result = ocr->ocr(*inMat, FLAGS_det, FLAGS_rec, FLAGS_cls);
EXE中正常，但在dll中出错，最后出错的地方是
ocr_det.cpp中的void DBDetector::Run
this->predictor_->Run();
2、报错是
0x000007FEAECC3391 (paddle_inference.dll)处(位于 test.exe 中)引发的异常: 0xC0000091: Floating-point overflow (参数: 

0x0000000000000000, 0x000000000000193A)。
所有图片和设置都和EXE的工程一样的。
3、临时方法。在Dll中把paddleocr.cpp
td::vector<OCRPredictResult> PPOCR::ocr(cv::Mat img, bool det, bool rec,
                                         bool cls) {

  std::vector<OCRPredictResult> ocr_result;
  // det
  //this->det(img, ocr_result);//dll中这个会出错,改成下页，图片中只有一行字时可以运行
  ocr_result.resize(1);
  ocr_result[0].box.resize(4);
  ocr_result[0].box[0].resize(2);
  ocr_result[0].box[0][0] = 0;
  ocr_result[0].box[0][1] = 0;
  
  ocr_result[0].box[1].resize(2);
  ocr_result[0].box[1][0] = img.cols;
  ocr_result[0].box[1][1] = 0;

  ocr_result[0].box[2].resize(2);
  ocr_result[0].box[2][0] = img.cols;
  ocr_result[0].box[2][1] = img.rows;

  ocr_result[0].box[3].resize(2);
  ocr_result[0].box[3][0] = 0;
  ocr_result[0].box[3][1] = img.rows;

  // crop image
  std::vector<cv::Mat> img_list;

4、临时方法问题。只能有一行文字，多行报错，无法识别报错。

**Comment by XiaoDongGuoGuo** [comment]
1、浮点溢出，检查下代码处理数字逻辑；
2、使用调试器（如 Visual Studio 的调试器）运行你的程序，当异常发生时，调试器会暂停执行，确定是哪个函数调用导致了异常。

**Comment by jwj76** [comment]
0x000007FEAECC3391 (paddle_inference.dll)处(位于 test.exe 中)引发的异常: 0xC0000091: Floating-point overflow (参数:

0x0000000000000000, 0x000000000000193A)。就是调试器报错信息。我跟踪了是ocr_det.cpp中的void DBDetector::Run
this->predictor_->Run();出错。可Run()是一个api调用。无法看到内部是什么问题

**Comment by jwj76** [comment]
Run()是paddle\include\paddle_inference_api.h定义的。
///
  /// \brief Run the prediction engine
  ///
  /// \return Whether the function executed successfully
  ///
  bool Run();

**Comment by jwj76** [comment]
找到问题了。是paddle_inference.dll版本的问题，换了一个就好了。

**Reply by running-az** [reply]
解决了吗？
